src/models/source_code_unit.rs

```diff
@@ -313,13 +313,6 @@ impl SourceCodeUnit {
     self.perform_delete_consecutive_new_lines();
   }
 
-  pub(crate) fn apply_edit(&mut self, edit: &Edit, parser: &mut Parser) -> InputEdit {
```

Comment:
inlined `apply_edit`, this simplifies the code and also kinda solves this bug further. 

Earlier we had a function called `apply_edit` which called `_apply_edit`  **and then** called `_delete_associated_comment` . Now `_delete_associated_comment` called `_apply_edit` (not **`apply_edit`** ), therefore `_delete_associated_comment` was not getting invoked recursively, and therefore we were not deleting a series of comments. 

I just inlined `apply_edit`  into `_apply_edit` and renamed `_apply_edit` to `apply_edit`. Now when `apply_edit` is called from `_delete_associated_comment`, it ensures that `_delete_associated_comment` is called recursively. (I adjusted all the signatures and simplified the code)